### PR TITLE
Noteworthy non-included storages: split base storages from others, NEO

### PR DESCRIPTION
**Pull request by jmuchemb**
In this PR, there's still a missing commit to add a entry for NEO. BTW, shall we sort alphabetically :)

### CODE REVIEW

File: doc/reference/storages.rst [hunk 1]
```diff
@@ -135,7 +135,7 @@ Demo storages are configured using the ``demostorage`` section::
 ``demostorage`` sections can contain up to 2 storage subsections,
 named ``base`` and ``changes``, specifying the demo storage's base and
 changes storages.  See :meth:`ZODB.DemoStorage.DemoStorage.__init__`
-for more on the base anc changes storages.
+for more on the base and changes storages.
```
Comment (jmuchemb) [inline doc/reference/storages.rst]: ... for more *information* on ... ?

Comment (jimfulton) [inline doc/reference/storages.rst]: See ZODB.DemoStorage.DemoStorage.__init__
for more on the base and changes storages.

**Comment by jimfulton** [comment]
I'm fine with sorting alphabetically. :)

**Comment by jmuchemb** [comment]
For me, it can be merged.

Maybe all those "For more information, see <url>" are too much. We could either remove them (because each first occurrence of the storage name points to the same url), or replace by

    homepage: <url>

File: doc/reference/storages.rst [hunk 2]
```diff
@@ -144,31 +144,46 @@ Options:
...
+-------------
+
+Unlike the included storages, all the implementations listed in this section
+allow multiple processes to share the same database.
+
+NEO
+  `NEO <https://lab.nexedi.com/nexedi/neoppod>`_ can spread data among several
+  computers for load-balancing and multi-master replication. It also supports
+  asynchronous replication to off-site NEO databases for further disaster
+  resistance without affecting local operation latency.
```
Comment (jamadden) [inline doc/reference/storages.rst]: I would consider it important to mention that of all the storages mentioned here, NEO is the only one that's Linux only; all the rest of them are operating-system independent. (The linux-only requirement is the major reason my organization can't test NEO---we develop on OS X.)

Comment (jmuchemb) [inline doc/reference/storages.rst]: I'm not sure it's the place to start listing drawbacks of each solution. The main limitations (like linux-only but also no blob support) look visible enough on the linked page.

The use of `epoll()` is the only reason why it requires Linux. In 2 places:
  https://lab.nexedi.com/nexedi/neoppod/blob/master/neo/lib/event.py
  https://lab.nexedi.com/nexedi/neoppod/blob/master/neo/tests/threaded/__init__.py

On our side, it's the contrary: we only develop and run test suites on Linux. We're open to contributions but unless someone provides test bots for other OS, we could only maintain it if the alternative implementation is not OS-specific, like select or libevent.

I'll see with @vpelletier

Comment (jamadden) [inline doc/reference/storages.rst]: I wasn't trying to criticize that as a drawback, just highlight that it's a substantial difference between that storage and the others. But then again, so is blob support.

You are right, it is quite clear from the linked page. So I withdraw my suggestion (unless someone else still feels like it needs to be included).

Comment (jimfulton) [inline doc/reference/storages.rst]: I'm fine with the NEO text as is.

**Comment by jimfulton** [comment]
I assume this is complete now. LGTM.

**Pull request merged**